acceptance/tests/snapshot-agent/snapshot_agent_vault_test.go

```diff
@@ -0,0 +1,138 @@
+package snapshotagent
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	terratestLogger "github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/environment"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/vault"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestSnapshotAgent_Vault installs snapshot agent config with an embedded token as a Vault secret.
+// It then installs Consul with Vault as a secrets backend and verifies that snapshot files
+// are generated.
+// Currently, the token needs to be embedded in the snapshot agent config due to a Consul
+// bug that does not recognize the token for snapshot command being configured via
+// a command line arg or an environment variable.
+func TestSnapshotAgent_Vault(t *testing.T) {
+	cfg := suite.Config()
+	ctx := suite.Environment().DefaultContext(t)
+	kubectlOptions := ctx.KubectlOptions(t)
+	ns := kubectlOptions.Namespace
+
+	consulReleaseName := helpers.RandomName()
+	vaultReleaseName := helpers.RandomName()
+
+	vaultCluster := vault.NewVaultCluster(t, ctx, cfg, vaultReleaseName, nil)
+	vaultCluster.Create(t, ctx)
+	// Vault is now installed in the cluster.
+
+	// Now fetch the Vault client so we can create the policies and secrets.
+	vaultClient := vaultCluster.VaultClient(t)
+
+	// gossipKey := configureGossipVaultSecret(t, vaultClient)
+
+	vault.CreateConnectCAPolicy(t, vaultClient, "dc1")
+	if cfg.EnableEnterprise {
+		vault.ConfigureEnterpriseLicenseVaultSecret(t, vaultClient, cfg)
+	}
+
+	bootstrapToken := vault.ConfigureACLTokenVaultSecret(t, vaultClient, "bootstrap")
+
+	vault.ConfigureSnapshotAgentSecret(t, vaultClient, cfg, generateSnapshotAgentConfig(t, bootstrapToken))
+
+	serverPolicies := "gossip,connect-ca-dc1,server-cert-dc1,bootstrap-token"
+	if cfg.EnableEnterprise {
+		serverPolicies += ",license"
+	}
+	vault.ConfigureKubernetesAuthRole(t, vaultClient, consulReleaseName, ns, "kubernetes", "server", serverPolicies)
```

Comment:
These roles are for these components to be able to get their secrets out of vault right?